### DISCUSSION TITLE
Weird issue, it works on web but not on android.

### DISCUSSION THREAD

**Discussion by CatalystMonish** [capawesome-team/capacitor-firebase#470] [Q&A]
``````
import React, { useEffect, useState } from "react";
import { FirebaseFirestore } from '@capacitor-firebase/firestore';
import { storage } from '../firebase.js';  // Adjust the path to your Firebase config file
import { ref, getDownloadURL } from 'firebase/storage';  // Import ref and getDownloadURL
import logo from "../images/logo.png";

function SkillPill({ skillId }) {
  const [skill, setSkill] = useState({});
  const [imageURL, setImageURL] = useState(null);

  useEffect(() => {
    const fetchSkillData = async () => {
      try {
        const { snapshot } = await FirebaseFirestore.getDocument({
          reference: `skills/${skillId}`
        });

        const skillData = snapshot.data;  // Assuming data will always be present
        setSkill(skillData);

        console.log("SKILLS", skillData)
        console.log("SKILLS IMG", skillData.skillImage)
        if (skillData.skillImage) {
          const storageRef = ref(storage, skillData.skillImage);
          const url = await getDownloadURL(storageRef);
          setImageURL(url);

        }
      } catch (error) {
        console.error("Error fetching skill data:", error);
      }
    };

    fetchSkillData();
  }, [skillId]);

  return (
      <div className="flex w-fit items-center rounded-full bg-light py-[0.1875rem] pr-[0.5rem] px-[0.1875rem]">
        {/*{imageURL && (*/}
        {/*    <img*/}
        {/*        width="20"*/}
        {/*        height="20"*/}
        {/*        placeholder={logo}*/}
        {/*        src={imageURL}*/}
        {/*        className="h-[1.25rem] w-[1.25rem] rounded-full bg-white"*/}
        {/*    />*/}
        {/*)}*/}
        <p className="ml-[0.3125rem] font-lexend text-small font-small">
          {skill.skillName}
        </p>
      </div>
  );
}

export default SkillPill;
``````

so the log are as follows
on web:
SKILLS IMG skills/thumbnails/8ivi5_Python_logo_01.svg.png
SkillPill.jsx:21 SKILLS {skillName: 'Python', skillCategory: 'Tech', skillImage: 'skills/thumbnails/8ivi5_Python_logo_01.svg.png', skillDescription: 'Python is a high-level, interpreted, and general-p…anguage known for its simplicity and readability.'}

onadroid:
2023-10-18 14:44:07.740 10807-10807 Capacitor/Console       in....cecatalyst.sourcecatalystplus  I  File: http://192.168.1.213:5173/src/components/SkillPill.jsx?t=1697620445419 - Line 39 - Msg: SKILLS {skillDescription=Python is a high-level, interpreted, and general-purpose programming language known for its simplicity and readability., skillName=Python, skillImage=skills/thumbnails/8ivi5_Python_logo_01.svg.png, skillCategory=Tech}
2023-10-18 14:44:07.741 10807-10807 Capacitor/Console       in....cecatalyst.sourcecatalystplus  I  File: http://192.168.1.213:5173/src/components/SkillPill.jsx?t=1697620445419 - Line 40 - Msg: SKILLSIMG undefined`

I tried different ways to access the object idk what I am missing
Msg: SKILLSIMG undefined`
not able to get this from skillData

**Comment by robingenz** [comment]
> console.log("SKILLS", skillData)

What is the output here on Android?

**Reply by CatalystMonish** [reply]
SKILLS {skillDescription=Python is a high-level, interpreted, and general-purpose programming language known for its simplicity and readability., skillName=Python, skillImage=skills/thumbnails/8ivi5_Python_logo_01.svg.png, skillCategory=Tech}
2023-10-18 14:44:07.741 10807-10807 Capacitor/Console in....cecatalyst.sourcecatalystplus I 

Line 40 - Msg: SKILLSIMG undefined`

**Reply by robingenz** [reply]
Can you share a screenshot of the logs in Chrome?

**Reply by CatalystMonish** [reply]
![image](https://github.com/capawesome-team/capacitor-firebase/assets/30138405/24f84e27-4ce7-41d8-9362-cbd75ec889e8)

**Reply by robingenz** [reply]
`SKILLS IMG` seems to work now.

**Reply by CatalystMonish** [reply]
Yes it works in chrome, the logcat in android it comes up as undefined
I tried various workarounds too, I tried stringifying the data to access it as a json object but no luck

**Reply by robingenz** [reply]
Okay, you misunderstood me. ^^ Please share a screenshot of the **Android** logs in Chrome. 
For more information, see https://ionicframework.com/docs/troubleshooting/debugging#android-and-chrome

**Reply by CatalystMonish** [reply]
I apologize, I didn't even know this was an option, but here you go 
![image](https://github.com/capawesome-team/capacitor-firebase/assets/30138405/26f524ec-ee38-46b7-a110-2f2aa304e84a)

**Reply by robingenz** [reply]
Okay, the issue is that the data isn't json but just a string. It should work if you parse it to json. But I recommend to delete the document and to create a new one with the data formatted as json.

**Reply by CatalystMonish** [reply]
Okay yeah I just noticed that,

on web the returned data is 
```{skillCategory: 'Tech', skillName: 'Python', skillImage: 'skills/thumbnails/8ivi5_Python_logo_01.svg.png', skillDescription: 'Python is a high-level, interpreted, and general-p…anguage known for its simplicity and readability.'}```
which is in json

while on android it is
```{skillDescription=Python is a high-level, interpreted, and general-purpose programming language known for its simplicity and readability., skillName=Python, skillImage=skills/thumbnails/8ivi5_Python_logo_01.svg.png, skillCategory=Tech}```

which has = inplace of  : and missing the '', why is this so? When the code is exactly the same? 

I think the document is in proper format, becuase its a old database, I am porting the app to use capacitor now.

**Reply by CatalystMonish** [reply]
````       const dataString = snapshot.data;
        const keyValuePairs = dataString.slice(1, -1).split(', ').map(pair => pair.split('='));
        const skillData = Object.fromEntries(keyValuePairs);
        console.log("SKILLS IMG", skillData.skillImage);

        setSkill(skillData);
````
        
When I do this, it works on android :), but then it breaks on web :(

**Reply by robingenz** [reply]
> `{skillDescription=Python is a high-level, interpreted, and general-purpose programming language known for its simplicity and readability., skillName=Python, skillImage=skills/thumbnails/8ivi5_Python_logo_01.svg.png, skillCategory=Tech}`

I see, this is not even valid json (see https://jsonlint.com/).
TBH,  you should fix the format of your saved data first. Then it will work on both platforms without any hacks.
If this is not an option, you can execute code with [`Capacitor.getPlatform()`](https://capacitorjs.com/docs/core-apis/web#getplatform) depending on the platform.

**Comment by CatalystMonish** [comment]
for now, I did use Capacitor.getPlatform(), marking this as closed, will update this if I find the root cause of this problem, Thanks @robingenz for your prompt replies with references to all the resources I needed.
